pkg/apis/certmanager/v1/types_certificate.go

```diff
@@ -168,7 +168,7 @@ type CertificateSpec struct {
 	// Minimum accepted value is 5 minutes.
 	// Value must be in units accepted by Go time.ParseDuration https://golang.org/pkg/time/#ParseDuration.
 	// +optional
```

Comment:
Let's not introduce `+kubebuilder:validation:Pattern` as part of this PR.
We have not yet used that pattern in cert-manager. Also, I'm not sure it will work with the generators that we use.
The webhooks will provide all required validation logic.

Reply:
I pushed a change to the update the `RenewBefore` GoDoc with the new behavior. Happy to rephrase if desired.

Reply:
Thanks! Very well explained description.